src/Libraries/Microsoft.Extensions.Http.Resilience/Resilience/HttpRequestMessageExtensions.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net.Http;
+using Microsoft.Shared.Diagnostics;
+using Polly;
+
+namespace Microsoft.Extensions.Http.Resilience;
+
+/// <summary>
+/// The resilience extensions for <see cref="HttpRequestMessage"/>.
+/// </summary>
+public static class HttpRequestMessageExtensions
+{
+#if !NET8_0_OR_GREATER
+    private const string ResilienceContextKey = "Resilience.Http.ResilienceContext";
+#else
+    private static readonly HttpRequestOptionsKey<ResilienceContext> _resilienceContextKey = new("Resilience.Http.ResilienceContext");
+#endif
+
+    /// <summary>
+    /// Gets the <see cref="ResilienceContext"/> from the request message.
+    /// </summary>
+    /// <param name="requestMessage">The request.</param>
+    /// <returns>An instance of <see cref="ResilienceContext"/> or <see langword="null"/>.</returns>
+    public static ResilienceContext? GetResilienceContext(this HttpRequestMessage requestMessage)
+    {
+        _ = Throw.IfNull(requestMessage);
+
+#if !NET8_0_OR_GREATER
```

Comment:
I think it would be nicer to invert this condition

Reply:
Done.